ner_v1/detectors/base_detector.py

```diff
@@ -106,8 +106,13 @@ def detect_bulk(self, messages=None, **kwargs):
         texts = messages
         entities_list, original_texts_list = self.detect_entity_bulk(texts=texts)
 
-        if entities_list:
+        fallback_value_list = kwargs.get('fallback_value')
+        if entities_list and not fallback_value_list:
             values_list, method, original_texts_list = entities_list, FROM_MESSAGE, original_texts_list
+        elif fallback_value_list:
```

Comment:
Fallback should be assigned if main text detection fails and if fallback is available